do_with_in_base/src/lib.rs

```diff
@@ -2950,6 +2950,60 @@ pub fn runMarkersHandler<T: StartMarker + Clone>(c: Configuration<T>, v: Variabl
   }
   Ok((v_out, quote!{}))
 }
+
+/// Provides tools for working with arrays.
+/// 
+/// *Syntax*: `$(array <q>? <command> <arguments>)`
+/// 
+/// An array in `do_with_in!` is a Group (delimited token stream) containing zero or more Groups, and nothing else.
+/// The array commands simply manipulate these sequences of tokens to produce an evaluated result.
+/// All commands that "modify" an array in fact produce a new evaluated result.
+/// By convention, arrays are delimited by square brackets e.g. `[{100} {a b}]`.
+/// 
+/// ## q mode ##
+/// 
+/// If the `array` token is followed immediately by a `q`, the command will treat its arguments as quoted and handle them as such.
+/// For example, `%(array q length %(quote {{1}}))` returns 1.
```

Comment:
Maybe example like `%(array q length %(quote [ {foo} ]))` evaluates to 1; `%(let x = { quux }) %(array ith 2 [ {1} {foo bar baz} { $x } ])` evaluates to `{ quux }`, where `%(let x = { quux }) %(array q ith 2 %(quote [ {1} {foo bar baz} { $x } ]))` evaluates to `%(quote { $x })`.